desktop_version/src/Map.cpp

```diff
@@ -2215,3 +2216,131 @@ void mapclass::twoframedelayfix(void)
     game.setstatedelay(0);
     script.load(game.newscript);
 }
+
+MapRenderData mapclass::get_render_data(void)
+{
+    MapRenderData data;
+    data.width = getwidth();
+    data.height = getheight();
+
+    data.startx = 0;
+    data.starty = 0;
+
+    // Region handling
+    if (region[currentregion].isvalid)
+    {
+        data.startx = region[currentregion].rx;
+        data.starty = region[currentregion].ry;
+        data.width = ((region[currentregion].rx2 - data.startx) + 1);
+        data.height = ((region[currentregion].ry2 - data.starty) + 1);
+    }
+
+    data.zoom = 1;
+
+    if (data.width <= 10 && data.height <= 10)
+    {
+        data.zoom = 2;
+    }
+    if (data.width <= 5 && data.height <= 5)
+    {
+        data.zoom = 4;
+    }
+
+    data.xoff = 0;
+    data.yoff = 0;
+
+    // Set minimap offsets
+    switch (data.zoom)
+    {
+    case 4:
+        data.xoff = 24 * (5 - data.width);
+        data.yoff = 18 * (5 - data.height);
+        break;
+    case 2:
+        data.xoff = 12 * (10 - data.width);
+        data.yoff = 9 * (10 - data.height);
+        break;
+    default:
+        data.xoff = 6 * (20 - data.width);
+        data.yoff = (int)(4.5 * (20 - data.height));
+        break;
+    }
+
+    data.pixelsx = 240 - (data.xoff * 2);
+    data.pixelsy = 180 - (data.yoff * 2);
+
+    data.legendxoff = 40 + data.xoff;
+    data.legendyoff = 21 + data.yoff;
+
+    // Magic numbers for centering legend tiles.
+    switch (data.zoom)
+    {
+    case 4:
+        data.legendxoff += 21;
+        data.legendyoff += 16;
+        break;
+    case 2:
+        data.legendxoff += 9;
+        data.legendyoff += 5;
+        break;
+    default:
+        data.legendxoff += 3;
+        data.legendyoff += 1;
+        break;
+    }
+
+    return data;
+}
+
+void mapclass::setregion(int id, int rx, int ry, int rx2, int ry2)
+{
+    if (INBOUNDS_ARR(id, region) && id > 0)
+    {
+        // swap the variables if they're entered in the wrong order
+        if (rx2 > rx)
```

Comment:
This is the wrong way around and not only does it not fix wrongly-ordered inputs, it actually breaks all correctly-ordered inputs by incorrectly swapping them around.

```suggestion
        if (rx2 < rx)
```